src/scarabee/_scarabee/moc_driver.cpp

```diff
@@ -333,7 +355,15 @@ void MOCDriver::solve_isotropic() {
       }
     }
 
-    if (mode_ == SimulationMode::Keff) {
+    if (mode_ == SimulationMode::Keff && !cmfd_) {
+      prev_keff = keff_;
+      keff_ = calc_keff(next_flux, flux_);
+      rel_diff_keff = std::abs(keff_ - prev_keff) / keff_;
+    }
+
+    // If MOC iterations are skipped compute Keff
+    // the normal way
+    if (mode_ == SimulationMode::Keff && cmfd_ && !cmfd_->solved()) {
```

Comment:
These two if statements do exactly the same thing. Please try to change the logic so that they are together instead of duplicating this code.

Reply:
So I believe the following logic will work to combine them into a single if statement:

```
if (mode_ == SimulationMode::Keff && 
    (cmfd_ == nullptr  || (cmfd_ != nullptr && cmfd_->solved() == false))) {
    prev_keff = keff_;
    keff_ = calc_keff(next_flux, flux_);
    rel_diff_keff = std::abs(keff_ - prev_keff) / keff_;
}
```